In [2]:
import pandas as pd
info_df=pd.read_csv("titanic_cleaned.csv")
info_df


,Passenger_ID,Survived,Passenger_Class,Name,Gender,Age,Sibling_Spouse,Parent_Children,Tickets,Fare,Port_Embarked,FAre
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A5 21171,7.25,Southampton,7.25
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.28,Cherbourg,71.28
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STONO2 3101282,7.92,Southampton,7.92
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.10,Southampton,53.10
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.05,Southampton,8.05
...,...,...,...,...,...,...,...,...,...,...,...,...
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.00,Southampton,13.00
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.00,Southampton,30.00
888,889,0,3,"Johnston, Miss. Catherine Helen ""Carrie""",female,28.0,1,2,WC 6607,23.45,Southampton,23.45
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.00,Cherbourg,30.00


In [12]:
info_df= info_df.drop(info_df.columns[-1], axis=1)


In [13]:
info_df.tail(5)

,Passenger_ID,Survived,Passenger_Class,Name,Gender,Age,Sibling_Spouse,Parent_Children,Tickets,Fare,Port_Embarked,Family_size,Age_Category
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.00,Southampton,1,Adult
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.00,Southampton,1,Adult
888,889,0,3,"Johnston, Miss. Catherine Helen ""Carrie""",female,28.0,1,2,WC 6607,23.45,Southampton,4,Adult
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.00,Cherbourg,1,Adult
890,891,0,3,"Dooley, Mr. Patrick",male,32.0,0,0,370376,7.75,Queenstown,1,Adult


In [15]:
info_df.shape

(891, 13)

In [16]:
info_df.columns

Index(['Passenger_ID', 'Survived', 'Passenger_Class', 'Name', 'Gender', 'Age',
       'Sibling_Spouse', 'Parent_Children', 'Tickets', 'Fare', 'Port_Embarked',
       'Family_size', 'Age_Category'],
      dtype='object')

In [17]:
info_df.describe()

,Passenger_ID,Survived,Passenger_Class,Age,Sibling_Spouse,Parent_Children,Fare,Family_size
count,891.000000,891.000000,891.000000,891.000000,891.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.361582,0.523008,0.381594,32.204366,1.904602
std,257.353842,0.486592,0.836071,13.019697,1.102743,0.806057,49.693414,1.613459
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000,1.000000
25%,223.500000,0.000000,2.000000,22.000000,0.000000,0.000000,7.910000,1.000000
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.450000,1.000000
75%,668.500000,1.000000,3.000000,35.000000,1.000000,0.000000,31.000000,2.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.330000,11.000000


In [18]:
info_df.duplicated().sum()

np.int64(0)

# Dataset Understanding
- This dataset is about titanic ship which faced a trouble and this dataset contain lot of columns which made people to get insights regarding that event
- This dataset contains 891 rows and 12 columns
- 

# Part 2: Feature Engineering


In [19]:
info_df.head(10)

,Passenger_ID,Survived,Passenger_Class,Name,Gender,Age,Sibling_Spouse,Parent_Children,Tickets,Fare,Port_Embarked,Family_size,Age_Category
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A5 21171,7.25,Southampton,2,Adult
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.28,Cherbourg,2,Adult
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STONO2 3101282,7.92,Southampton,1,Adult
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.10,Southampton,2,Adult
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.05,Southampton,1,Adult
5,6,0,3,"Moran, Mr. James",male,28.0,0,0,330877,8.46,Queenstown,1,Adult
6,7,0,1,"McCarthy, Mr. Timothy J",male,54.0,0,0,17463,51.86,Southampton,1,Adult
7,8,0,3,"Palsson, Master. Gosta Leonard",male,2.0,3,1,349909,21.08,Southampton,5,child
8,9,1,3,"Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)",female,27.0,0,2,347742,11.13,Southampton,3,Adult
9,10,1,2,"Nasser, Mrs. Nicholas (Adele Achem)",female,14.0,1,0,237736,30.07,Cherbourg,2,Teen


In [20]:
info_df["Family_size"]=info_df["Sibling_Spouse"]+info_df["Parent_Children"]+1


In [21]:
info_df

,Passenger_ID,Survived,Passenger_Class,Name,Gender,Age,Sibling_Spouse,Parent_Children,Tickets,Fare,Port_Embarked,Family_size,Age_Category
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A5 21171,7.25,Southampton,2,Adult
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.28,Cherbourg,2,Adult
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STONO2 3101282,7.92,Southampton,1,Adult
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.10,Southampton,2,Adult
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.05,Southampton,1,Adult
...,...,...,...,...,...,...,...,...,...,...,...,...,...
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.00,Southampton,1,Adult
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.00,Southampton,1,Adult
888,889,0,3,"Johnston, Miss. Catherine Helen ""Carrie""",female,28.0,1,2,WC 6607,23.45,Southampton,4,Adult
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.00,Cherbourg,1,Adult


In [22]:
#making column of age group
def age_group(age):
    if age<12:
        return "child"
    elif age<18:
        return "Teen"
    elif age<65:
        return "Adult"
    else:
        return "Old_Age"
info_df["Age_Category"]=info_df["Age"].apply(age_group)
#making column of fare category
def fare_distribution(fare):
    if fare<200:
        return "Low"
    elif fare>200 and fare<400:
        return "Medium"
    else:
        return "High"
info_df["Fare_Category"]=info_df["Fare"].apply(fare_distribution)
info_df

,Passenger_ID,Survived,Passenger_Class,Name,Gender,Age,Sibling_Spouse,Parent_Children,Tickets,Fare,Port_Embarked,Family_size,Age_Category,Fare_Category
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A5 21171,7.25,Southampton,2,Adult,Low
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.28,Cherbourg,2,Adult,Low
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STONO2 3101282,7.92,Southampton,1,Adult,Low
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.10,Southampton,2,Adult,Low
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.05,Southampton,1,Adult,Low
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.00,Southampton,1,Adult,Low
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.00,Southampton,1,Adult,Low
888,889,0,3,"Johnston, Miss. Catherine Helen ""Carrie""",female,28.0,1,2,WC 6607,23.45,Southampton,4,Adult,Low
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.00,Cherbourg,1,Adult,Low
